### PR TITLE
flatten nested objects to top level

### PR DESCRIPTION
**Pull request by hankim813**
Kissmetrics wants nested objects to be flattened before sending them as event properties or user traits.

This PR will have this behavior:

``` json
{
  "input": {
    "type": "track",
    "userId": "hanhan",
    "event": "some event",
    "timestamp": "2014",
    "properties": {
      "revenue": 19.99,
      "some-property": true,
      "time": "2014-01-01",
      "list": [1, 2, 3],
      "school": {
        "middle": { "state": "RI" },
        "college": { "state": "MA" },
        "jobs": ["segment", "nasa"]
      }
    }
  },
  "output": {
    "Billing Amount": 19.99,
    "some-property": "true",
    "revenue": "19.99",
    "time": 1388534400,
    "list": "1,2,3",
    "school.middle.state": "RI",
    "school.college.state": "MA",
    "school.jobs": "segment,nasa",
    "_p": "hanhan",
    "_t": 1388534400,
    "_k": "2b93bdf937c54fc7da2b5c6015c273bf3919c273",
    "_n": "some event",
    "_d": 1
  }
}
```

@amillet89

### CODE REVIEW

**Comment by carlsednaoui** [comment]
Hi @hankim813 curious to know if we could merge this PR today. Would love to get this KM bug fix in place :)

**Comment by hankim813** [comment]
@carlsednaoui hey Carl -- server side is ready but we're figuring out a way to make this same fix for the client side as well so that we have parity for the integration as a whole. trying to get the client side stuff pushed out too so hopefully everything will be ready to go by end of this weekend or early monday!

**Comment by carlsednaoui** [comment]
Awesome, thanks Han!

On Friday, October 23, 2015, Han notifications@github.com wrote:

> @carlsednaoui https://github.com/carlsednaoui hey Carl -- server side
> is ready but we're figuring out a way to make this same fix for the client
> side as well so that we have parity for the integration as a whole. trying
> to get the client side stuff pushed out too so hopefully everything will be
> ready to go by end of this weekend or early monday!
> 
> —
> Reply to this email directly or view it on GitHub
> https://github.com/segmentio/integration-kissmetrics/pull/12#issuecomment-150711122
> .

## 

Mobile

**Comment by hankim813** [comment]
@carlsednaoui hey i just finished the client side PR, hoping to deploy both this PR and that today. just going to have them reviewed a bit more

File: lib/mapper.js [hunk 1]
```diff
@@ -53,7 +54,6 @@ exports.track = function(track) {
   return extend(properties, {
     _p: track.userId() || track.sessionId(),
     _t: toUnixTimestamp(track.timestamp()),
-    _k: this.settings.apiKey,
```
Comment (f2prateek) [inline lib/mapper.js:56]: Why was this removed?

File: test/index.js [hunk 2]
```diff
@@ -86,6 +86,16 @@ describe('KISSmetrics', function () {
       kissmetrics.track(track, done);
     });
 
+    it('should properly set nested objects as properties', function(done){
+      var json = test.fixture('track-clean');
```
Comment (f2prateek) [inline test/index.js:94]: Also needs a mapper test for the `track-clean` fixture.

**Comment by carlsednaoui** [comment]
Thanks for the update, @hankim813.

File: lib/mapper.js (hunk 1 above)
Comment (hankim813) [inline lib/mapper.js:56]: @f2prateek we're already appending that key in the through the `createBaseTrackProperties` function.

**Comment by hankim813** [comment]
@carlsednaoui shipping all the KM stuff right now!

**Comment by carlsednaoui** [comment]
Wahoo! Thanks @hankim813
